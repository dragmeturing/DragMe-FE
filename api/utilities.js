export const BEUrl = 'https://dragme-be.herokuapp.com/api';