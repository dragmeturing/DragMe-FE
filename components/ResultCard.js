import React from 'react';
import { TouchableHighlight, Text, View, ImageBackground, StyleSheet } from 'react-native';
import { cleanDate, cleanTime } from './helper';
import { withNavigation } from "react-navigation";

const ResultCard = (props) => {
  const { id, name, venue_name, poster_url } = props.data;
  const { resultText, card, header, textHolder } = localStyles;
  const date = cleanDate(props.data.date);
  const time = cleanTime(props.data.date);
  return (
    <TouchableHighlight
      style={card}
      onPress={() => props.navigation.navigate("Show", { id })}
    >
      <ImageBackground
        source={{ uri: poster_url }}
        style={{ width: "100%", height: "100%" }}
        imageStyle={{ opacity: 0.7 }}
        blurRadius={7}
      >
        <View style={textHolder}>
          <Text style={[resultText, header]}>{name}</Text>
          <Text style={[resultText]}>
            {date} - {time}
          </Text>
          <Text style={[resultText]}>{venue_name}</Text>
        </View>
      </ImageBackground>
    </TouchableHighlight>
  );
};

const localStyles = StyleSheet.create({
  resultText: {
    color: "white",
    textAlign: "center",
    fontSize: 24
  },
  header: {
    fontSize: 32
  },
  card: {
    marginVertical: 3,
    height: 150,
    backgroundColor: "black"
  },
  textHolder: {
    display: "flex",
    height: "100%",
    justifyContent: "space-around",
    padding: 10
  }
});
  
export default withNavigation(ResultCard);