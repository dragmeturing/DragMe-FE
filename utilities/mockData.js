export const shows = [
  {
    id: 1,
    name: "Drag Roulette Show",
    description:
      "Your Friday LATE NIGHT at Mile High Hamburger Mary’s has been upgraded to hysterical! Every 1st and 3rd Friday!",
    event_url:
      "https://www.hamburgermarys.com/denver/events/event/drag-roulette-show/",
    date: "2019-07-14 15:16:38.643787",
    poster_url:
      "https://www.hamburgermarys.com/denver/wp/wp-content/uploads/2019/02/drag-roulette-1024x745.jpg",
    venue_name: "Hamburger Mary's Bar & Grille",
    venue_google_id: "1234556"
  },
  {
    id: 2,
    name: "Drag Roulette Show",
    description:
      "Your Friday LATE NIGHT at Mile High Hamburger Mary’s has been upgraded to hysterical! Every 1st and 3rd Friday!",
    event_url:
      "https://www.hamburgermarys.com/denver/events/event/drag-roulette-show/",
    date: "2019-07-14 15:16:38.643787",
    poster_url:
      "https://www.hamburgermarys.com/denver/wp/wp-content/uploads/2019/02/drag-roulette-1024x745.jpg",
    venue_name: "Hamburger Mary's Bar & Grille",
    venue_google_id: "1234556"
  },
  {
    id: 3,
    name: "Drag Roulette Show",
    description:
      "Your Friday LATE NIGHT at Mile High Hamburger Mary’s has been upgraded to hysterical! Every 1st and 3rd Friday!",
    event_url:
      "https://www.hamburgermarys.com/denver/events/event/drag-roulette-show/",
    date: "2019-07-14 15:16:38.643787",
    poster_url:
      "https://www.hamburgermarys.com/denver/wp/wp-content/uploads/2019/02/drag-roulette-1024x745.jpg",
    venue_name: "Hamburger Mary's Bar & Grille",
    venue_google_id: "1234556"
  }
];

export const venues = [
  {
    id: 1,
    venue_name: "The Clocktower Cabaret",
    venue_google_id: "ChIJASbMldp4bIcRiFeLsW3CUf4"
  },
  {
    id: 2,
    venue_name: "Mile High Hamburger Mary's",
    venue_google_id: "ChIJL4ZMpzN5bIcRe_ttaF-7N3E"
  },
  {
    id: 3,
    venue_name: "Tracks",
    venue_google_id: "ChIJU0l3jxp5bIcRcOzzQKHMBcM"
  },
  {
    id: 4,
    venue_name: "Charlie's Nightclub Denver",
    venue_google_id: "ChIJydsjLC15bIcRcnnY3EIbTgs"
  },
  {
    id: 5,
    venue_name: "Blush & Blu",
    venue_google_id: "ChIJ445gusx-bIcRDpB_himzcuY"
  },
  {
    id: 6,
    venue_name: "X BAR",
    venue_google_id: "ChIJw4sroyx5bIcR5jU8EDoneGU"
  },
  {
    id: 7,
    venue_name: "Pride and Swagger",
    venue_google_id: "ChIJCR82wSt5bIcRsu6dJbKpoX8"
  },
  {
    id: 8,
    venue_name: "Gladys",
    venue_google_id: "ChIJR9gq2Tp_bIcRrtd8dA47as8"
  },
  {
    id: 9,
    venue_name: "#VYBE",
    venue_google_id: "ChIJs4T9jil_bIcRCcy9CMKwJAA"
  }
];
