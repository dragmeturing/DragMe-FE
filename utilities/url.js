export const BEurl = 'insert AWS URL';
export const googleRoot = "https://maps.googleapis.com/maps/api/";
export const uploadImageURL = "https://dragmeuploadimage.herokuapp.com";

