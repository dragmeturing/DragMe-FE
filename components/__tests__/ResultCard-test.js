import React from 'react';
import { shallow } from 'enzyme';
import ResultCard from '../ResultCard';
import { shows } from '../../utilities/mockData';

describe('ResultCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ResultCard data={shows[0]}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
