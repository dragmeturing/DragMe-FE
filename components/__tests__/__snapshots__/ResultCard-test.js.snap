// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard should match the snapshot 1`] = `ShallowWrapper {}`;
